TEE.md

```diff
@@ -0,0 +1,79 @@
+- Hard Launch: node operators will be required to run their node inside a TEE.
+
+In order to protect the network from the worst case scenario (complete loss of funds due to loss of key shares or an incapacitated network), a disaster recovery plan is prepared.
+
+
+## Disaster Recovery
+Distaster recovery is a plan intended to prevent a permanent loss of the signing quorum.
+As long as the secret shares of the MPC nodes are securely backed-up outside of the TEE environment in which the node is running, it highly likely that the network will be able to recover from otherwise catastrophic events.
+
+Therefore, the disaster recovery plan aims at establishing a secure mechanism allowing to back-up the secret shares held by the nodes. It encompasses two steps:
+1. (Back-up Generation): export the private keys from the TEE in a secure manner.
+2. (Recovery): when required, securely import the private keys into the node.
+
+### Back-up mechanism:
+On a high-level, the back-up mechanism works like this:
+
+1. The node running inside the TEE and the back-up recipient must agree on a key used to encrypt the secrets.
+2. The node encrypts the sensitive secret shares with the derived key.
+3. The node exports the encrypted secret shares.
+
+#### Agreeing on an encryption key
+A pair-wise key establishement scheme can be leveraged to establish a symmetric key. A suitable implementation can be found in [libsodium](https://doc.libsodium.org/) and its rust wrapper [libsodium-rs](https://docs.rs/crate/libsodium-rs/latest). Specifically, the **key exchange protocol based on X25519** would be a good fit for the given set-up (c.f. [libsodium](https://doc.libsodium.org/key_exchange) and [libsodium-rs](https://docs.rs/libsodium-rs/latest/libsodium_rs/crypto_kx/index.html)).
+
+_Note: This key exchange protocol is on Curve25519, which is different from the Edwards curve used by NEAR._
+
+The key establishment scheme requires that the node as well as the back-up recipient have *mutually authenticated Curve25519 public keys*. The NEAR blockchain can be leveraged for this, more specifically:
+- the node operator whitelists a NEAR-account associated to the back-up recipient in the MPC smart contract.
+- the back-up recipient submits a public key (Curve25519) to the MPC smart contract. The smart contract only accepts it if the account has been whitelisted by the node operator.
+- the node submits a public key (Curve25519) to the MPC smart contract. The smart contract only accepts it if the public key submitting this transaction is an access key of the node operators account.
+
+_Note: Curve25519 and Edwards curve are [birationally equivalent](https://crypto.stackexchange.com/questions/43013/what-does-birational-equivalence-mean-in-a-cryptographic-context), so one could theoretically use NEAR account keys. But it seems cleaner if we just have the node and the back-up recipient generate dedicated keys for this._
```

Comment:
Isn't Edwards curves a family of curves[^1]? Which one do we refer to here? Afaik you can define Edwards curve's over many different fields.

[^1]: https://en.wikipedia.org/wiki/Edwards_curve

Reply:
good point! https://github.com/near/mpc/pull/877/commits/731460722abfeb726d5c1ae58aa5f46c104c4e88